test/powershell/engine/Module/TestModuleManifest.Tests.ps1

```diff
@@ -217,3 +217,30 @@ Describe "Tests for circular references in required modules" -tags "CI" {
         { TestImportModule $false $false $true } | ShouldBeErrorId "Modules_InvalidManifest,Microsoft.PowerShell.Commands.ImportModuleCommand"
     }
 }
+
+Describe "Test-ModuleManifest Performance bug followup" -tags "CI" {
+    BeforeAll {
+        $TestModulesFolder= 'testmodulerunspace'
+        $TestModulesPath = Join-path "$PSScriptRoot\assets" $TestModulesFolder
+        $UserModulesPath = Join-path "$pshome\Modules"
+
+        # Install the Test Module
+        Copy-Item $TestModulesPath\* $UserModulesPath -Recurse -Force
+    }
+
+    It "Test-ModuleManifest should not load unnessary modules" {
+
+        $job = start-job -name "job1" -ScriptBlock {test-modulemanifest "$using:UserModulesPath\ModuleWithDependencies2\2.0\ModuleWithDependencies2.psd1" -verbose} | Wait-Job
+        
+        $verbose = $job.ChildJobs[0].Verbose.ReadAll()
+        $verbose.Count -le 15 | should be true
```

Comment:
And please put your rational about using the number `'15'` in comments here, so that people look at this code later will understand it.